### ISSUE TITLE
Support TSDB metadata version 2

### ISSUE THREAD

**Issue by mattbostock** [improbable-eng/thanos#239]
Prometheus 2.1.0 incremented the TSDB metadata version number:
https://github.com/prometheus/prometheus/releases/tag/v2.1.0

...which prevents the sidecar from working with Prometheus versions 2.1.0 and above:

```
level=warn name=sidecar-3 ts=2018-03-06T23:26:08.969876206Z caller=shipper.go:238 msg="reading meta file failed" err="unexpected meta file version 2"
```

**Comment by Bplotka** [comment]
Please don't use 2.1.0 it's broken. Especially for thanos. Please use `2.2-rc` or `2.0` ):

**Comment by Bplotka** [comment]
This is quite important -> there was a bug that bumbs meta file version (unnecessary) but not index file version, and index was actually changed. 

I would also remove all blocks created for 2.1.0 for sanity. (But I think @fabxc implemented repairing step for these blocks in 2.2). We documented that we only support 2.0 for now, but maybe not clearly ): 

`2.2` should be fine!

**Comment by mattbostock** [comment]
Thanks @Bplotka. In this case I was using the latest Prometheus version available via Homebrew for local testing. Closing.
